src/main.cpp

```diff
@@ -3760,6 +3765,36 @@ std::string GetWarnings(const std::string& strFor)
 // Messages
 //
 
+static std::map<std::string, size_t> maxMessageSizes = boost::assign::map_list_of
+    ("addr", 2+1000*30)
+    ("getaddr",0)
+    ("getdata", 2+MAX_INV_SZ*36)
+    ("inv", 2+MAX_INV_SZ*36)
+    ("mempool",0)
+    ("notfound", 2+MAX_INV_SZ*36)
+    ("ping",8)
+    ("pong",8)
+    ("verack", 0)
+    ;
+
+bool static SanityCheckMessage(CNode* peer, const CNetMessage& msg)
+{
+    if (msg.hdr.nMessageSize > MAX_PROTOCOL_MESSAGE_LENGTH) {
+        LogPrint("net", "Oversized message from peer=%i, disconnecting", peer->GetId());
+        return false;
```

Comment:
Harmonize the two LogPrint() calls.  No reason to omit strCommand diagnostic output on this first LogPrint() here.